nebula_decoders/include/nebula_decoders/nebula_decoders_hesai/decoders/hesai_decoder.hpp

```diff
@@ -41,84 +40,73 @@ namespace nebula::drivers
-  {
-    distance_filtered_count = 0;
-    fov_filtered_count = 0;
-    timestamp_filtered_count = 0;
-    invalid_point_count = 0;
-    multiple_return_point_count = 0;
-    mutliple_return_point_count = 0;
-    total_kept_point_count = 0;
-    invalid_packet_count = 0;
-    cloud_distance_min_m = 0;
-    cloud_distance_max_m = 0;
-    cloud_azimuth_min_rad = 0;
-    cloud_azimuth_max_rad = 0;
-    packet_timestamp_min_ns = 0;
-    packet_timestamp_max_ns = 0;
-  }
+  uint64_t distance_filtered_count = 0;
+  uint64_t fov_filtered_count = 0;
+  uint64_t invalid_point_count = 0;
+  uint64_t identical_filtered_count = 0;
+  uint64_t multiple_return_filtered_count = 0;
+  uint64_t total_kept_point_count = 0;
+  uint64_t invalid_packet_count = 0;
+  float cloud_distance_min_m = std::numeric_limits<float>::infinity();
+  float cloud_distance_max_m = std::numeric_limits<float>::lowest();
+  float cloud_azimuth_min_deg = std::numeric_limits<float>::infinity();
+  float cloud_azimuth_max_rad = std::numeric_limits<float>::lowest();
+  uint64_t packet_timestamp_min_ns = std::numeric_limits<uint64_t>::max();
+  uint64_t packet_timestamp_max_ns = std::numeric_limits<uint64_t>::min();
 
   [[nodiscard]] nlohmann::ordered_json to_json() const
   {
     nlohmann::json distance_j;
-    distance_j["filter"] = "distance";
-    distance_j["distance_filtered_count"] = distance_filtered_count;
-    distance_j["cloud_distance_min_m"] = cloud_distance_min_m;
-    distance_j["cloud_distance_max_m"] = cloud_distance_max_m;
+    distance_j["name"] = "distance";
+    distance_j["filtered_count"] = distance_filtered_count;
+    // distance_j["cloud_distance_min_m"] = cloud_distance_min_m;
+    // distance_j["cloud_distance_max_m"] = cloud_distance_max_m;
     nlohmann::json fov_j;
-    fov_j["filter"] = "fov";
-    fov_j["fov_filtered_count"] = fov_filtered_count;
-    fov_j["cloud_azimuth_min_rad"] = cloud_azimuth_min_rad;
-    fov_j["cloud_azimuth_max_rad"] = cloud_azimuth_max_rad;
-    nlohmann::json timestamp_j;
-    timestamp_j["filter"] = "timestamp";
-    timestamp_j["timestamp_filtered_count"] = timestamp_filtered_count;
-    timestamp_j["packet_timestamp_min_ns"] = packet_timestamp_min_ns;
-    timestamp_j["packet_timestamp_max_ns"] = packet_timestamp_max_ns;
+    fov_j["name"] = "fov";
+    fov_j["filtered_count"] = fov_filtered_count;
+    // fov_j["cloud_azimuth_min_deg"] = cloud_azimuth_min_deg;
+    // fov_j["cloud_azimuth_max_rad"] = cloud_azimuth_max_rad;
+    nlohmann::json identical_j;
+    identical_j["name"] = "identical";
+    identical_j["filtered_count"] = identical_filtered_count;
+    nlohmann::json multiple_j;
+    multiple_j["name"] = "multiple";
+    multiple_j["filtered_count"] = multiple_return_filtered_count;
     nlohmann::json invalid_j;
-    invalid_j["filter"] = "invalid";
+    invalid_j["name"] = "invalid";
     invalid_j["invalid_point_count"] = invalid_point_count;
     invalid_j["invalid_packet_count"] = invalid_packet_count;
```

Comment:
Invalid points and packets are different concepts:
* while invalid points are normal (the sensor sends them when there was no object hit, or when an object is too close),
* invalid packets are an error (the size does not match our expectations)

So, please make the invalid points as part of the filter pipeline, and move invalid packets to the top level.
Also see [this previous comment](https://github.com/tier4/nebula/pull/247#discussion_r1903699586).